lib/semantics/symbol.cc

```diff
@@ -586,4 +586,28 @@ void TypeParamDetails::set_type(const DeclTypeSpec &type) {
   type_ = &type;
 }
 
+bool GenericKind::IsIntrinsicOperator() const {
+  return Is(OtherKind::Concat) || Has<common::LogicalOperator>() ||
+      Has<common::NumericOperator>() || Has<common::RelationalOperator>();
+}
+
+bool GenericKind::IsOperator() const {
+  return IsDefinedOperator() || IsIntrinsicOperator();
+}
+
+std::string GenericKind::ToString() const {
+  return std::visit(
+      common::visitors{
+          [&](const OtherKind &x) { return EnumToString(x); },
+          [&](const DefinedIo &x) { return EnumToString(x); },
+          [&](const auto &x) { return common::EnumToString(x); },
```

Comment:
I think you can do without the reference captures on these lambdas.

Reply:
Right